Src/DependencyCollector/Shared.Tests/Implementation/EventHubsDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,168 @@
+                var telemetry  = this.TrackOperation<DependencyTelemetry>(listener, "Microsoft.Azure.EventHubs.Send", TaskStatus.RanToCompletion);
+
+                Assert.IsNotNull(telemetry);
+                Assert.AreEqual("Send", telemetry.Name);
+                Assert.AreEqual(RemoteDependencyConstants.AzureEventHubs, telemetry.Type);
+                Assert.AreEqual("sb://eventhubname.servicebus.windows.net/ | ehname", telemetry.Target);
+                Assert.IsTrue(telemetry.Success.Value);
+
+                Assert.AreEqual(parentActivity.Id, telemetry.Context.Operation.ParentId);
+                Assert.AreEqual(parentActivity.RootId, telemetry.Context.Operation.Id);
+                Assert.AreEqual("v1", telemetry.Properties["k1"]);
+                Assert.AreEqual("eventhubname.servicebus.windows.net", telemetry.Properties["peer.hostname"]);
+                Assert.AreEqual("ehname", telemetry.Properties["eh.event_hub_name"]);
+                Assert.AreEqual("SomePartitionKeyHere", telemetry.Properties["eh.partition_key"]);
+                Assert.AreEqual("EventHubClient1(ehname)", telemetry.Properties["eh.client_id"]);
+            }
+        }
+
+        [TestMethod]
+        public void EventHubsFailedSendIsHandled()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.IncludeDiagnosticSourceActivities.Add("Microsoft.Azure.EventHubs");
+                module.Initialize(this.configuration);
+
+                DiagnosticListener listener = new DiagnosticListener("Microsoft.Azure.EventHubs");
+
+                Activity parentActivity = new Activity("parent").AddBaggage("k1", "v1").Start();
+                var telemetry = this.TrackOperation<DependencyTelemetry>(listener, "Microsoft.Azure.EventHubs.Send", TaskStatus.Faulted);
+
+                Assert.IsNotNull(telemetry);
+                Assert.AreEqual("Send", telemetry.Name);
+                Assert.AreEqual(RemoteDependencyConstants.AzureEventHubs, telemetry.Type);
+                Assert.AreEqual("sb://eventhubname.servicebus.windows.net/ | ehname", telemetry.Target);
+                Assert.IsFalse(telemetry.Success.Value);
+
+                Assert.AreEqual(parentActivity.Id, telemetry.Context.Operation.ParentId);
+                Assert.AreEqual(parentActivity.RootId, telemetry.Context.Operation.Id);
+                Assert.AreEqual("v1", telemetry.Properties["k1"]);
+                Assert.AreEqual("eventhubname.servicebus.windows.net", telemetry.Properties["peer.hostname"]);
+                Assert.AreEqual("ehname", telemetry.Properties["eh.event_hub_name"]);
+                Assert.AreEqual("SomePartitionKeyHere", telemetry.Properties["eh.partition_key"]);
+                Assert.AreEqual("EventHubClient1(ehname)", telemetry.Properties["eh.client_id"]);
+            }
+        }
+
+        [TestMethod]
+        public void EventHubsSendExceptionsAreIgnored()
```

Comment:
Could you please kindly explain why is exception ignored? Thanks!

Reply:
Exceptions are usually well covered by user code or would be tracked in other context. The concern is that collecting a lot of transient exceptions would be noisy and incur telemetry cost

Reply:
Fair enough. I see that the implementation for S.B. shares the same logic. I totally have no problem with it. Thanks for letting me know the reason behind.